src/platform/desktop_notifications/desktopnotify.cpp

```diff
@@ -109,4 +110,29 @@ void DesktopNotify::notifyMessage(const NotificationData& notificationData)
     notify_notification_show(impl->lastNotification.get(), nullptr);
 }
 
-#endif
+#else // Q_OS_MAC || Q_OS_WIN
+
+struct DesktopNotify::Impl
+{
+    Impl() {}
+
+    ~Impl() {}
+};
+
+DesktopNotify::DesktopNotify(Settings& settings_)
+    : impl{std::make_unique<Impl>()}
+    , settings{settings_}
+{
+}
+
+DesktopNotify::~DesktopNotify() = default;
+
+void DesktopNotify::notifyMessage(const NotificationData& notificationData)
+{
+    if (!(settings.getNotify() && settings.getDesktopNotify())) {
```

Comment:
This code doesn't seem necessary. :P

Reply:
Why not? We want to give the user the ability to disable notifications at runtime.

Reply:
Because all it does is return, and then we'll return regardless 2 lines later, having done nothing?

Reply:
Ah, true. Well, it's outdated now, PTAL :).